ARCHITECTURE.md

```diff
@@ -0,0 +1,423 @@
+
+### Logging
+
+For logging, we use [winston] and [morgan]. Winston is set up outside of the main application files in [`config.js`](config.js), this allows us to inject a mock/silent logger in the [integration tests](#integration-tests).
+
+Morgan is used to log requests, and this is mounted as an Express middleware. We actually stream morgan's request logs into winston under the `verbose` level. This allows users to easily ignore them by tweaking their preferred log level.
+
+
+## Front End
+
+OK, so you want to know how to work on Pa11y Sidekick's front end? Let's do it! This section will explain how to add templates (with [Dust.js]), CSS (with [Sass]), and client-side JavaScript (with [Browserify]).
+
+### View Templates
+
+View templates live in the `view` folder, and are written in the [Dust.js] templating language. We make use of [blocks and inline partials][dust-blocks] for template inheritance, and try to split repeated code into partials. Here's an example view:
+
+```html
+<!-- Extend the default layout -->
+{>"layout/default"/}
+
+<!-- insert the following into the layout title block -->
+{< title}
+    My Page
+{/title}
+
+<!-- insert the following into the layout content block -->
+{< content}
+
+    <h1>Hello World!</h1>
+
+    <p>This is data from the controller: {example}</p>
+
+{/content}
+```
+
+In order to see your view rendered in-browser, a controller will have to reference it. [Take a look at the controller documentation](#controllers) for more information on how to do this.
+
+The `error.dust` view is a special one – it's used to render errors that get caught by the application.
+
+#### Layouts
+
+As mentioned above, we use [blocks and inline partials][dust-blocks] for template inheritance. This allows for multiple modifiable blocks in our layouts which each view can add data to.
+
+The [default layout](view/layout/default.dust) is used by most views; it provides boilerplate HTML and loads in the required scripts and styles.
+
+#### Partials
+
+Partials are used to reduce repetition in view files, and common markup patterns should be moved into partials. The best way to get aquainted with how we use partials is to [look through the folder](view/partial).
+
+#### Default View Data
+
+Controllers are normally in charge of passing data to the view, and a controller is required for both rendering the views and interfacing between them and the database. For data that's static and unchanging, you can use [`view/defaults.json`](view/defaults.json) – This file is loaded into every view.
+
+So if the default view data file contains this:
+
+```json
+{
+    "thing": "World"
+}
+```
+
+Then this view will render "Hello World!":
+
+```html
+Hello {thing}!
+```
+
+### View Helpers
+
+View helpers are used to extend the templating language, and to ensure that your views themselves don't have as much logic in them.
+
```

Comment:
Is it worth mentioning guidelines for view helpers? As in … "avoid writing unless absolutely necessary, try to get the data in the format you need before you resort to modifying the view"?